DigitalLearningSolutions.Data.Tests/Services/DelegateUploadFileServiceTests.cs

```diff
@@ -0,0 +1,713 @@
+                .MustNotHaveHappened();
+            A.CallTo(() => registrationDataService.RegisterDelegateByCentre(A<DelegateRegistrationModel>._))
+                .MustNotHaveHappened();
+        }
+
+        private void ShouldJustHaveOneError(BulkUploadResult result)
+        {
+            result.Processed.Should().Be(1);
+            result.Updated.Should().Be(0);
+            result.Registered.Should().Be(0);
+            result.Skipped.Should().Be(0);
+            result.Errors.Should().HaveCount(1);
+        }
+
+        [Test]
+        public void OpenDelegatesTable_returns_table_correctly()
+        {
+            // Given
+            var stream = File.OpenRead(
+                TestContext.CurrentContext.TestDirectory + "\\TestData\\DelegateUploadTest.xlsx"
+            );
+            var file = new FormFile(stream, 0, stream.Length, null, Path.GetFileName(stream.Name));
+
+            // When
+            var table = delegateUploadFileService.OpenDelegatesTable(file);
+
+            // Then
+            using (new AssertionScope())
+            {
+                var headers = table.Fields.Select(x => x.Name).ToList();
+                headers[0].Should().Be("LastName");
+                headers[1].Should().Be("FirstName");
+                headers[2].Should().Be("DelegateID");
+                headers[3].Should().Be("AliasID");
+                headers[4].Should().Be("JobGroupID");
+                headers[5].Should().Be("Answer1");
+                headers[6].Should().Be("Answer2");
+                headers[7].Should().Be("Answer3");
+                headers[8].Should().Be("Answer4");
+                headers[9].Should().Be("Answer5");
+                headers[10].Should().Be("Answer6");
+                headers[11].Should().Be("Active");
+                headers[12].Should().Be("EmailAddress");
+                table.RowCount().Should().Be(4);
+                var row = table.Row(2);
+                row.Cell(1).GetString().Should().Be("Person");
+                row.Cell(2).GetString().Should().Be("Fake");
+                row.Cell(3).GetString().Should().Be("TU67");
+                row.Cell(4).GetString().Should().BeEmpty();
+                row.Cell(5).GetString().Should().Be("1");
+                row.Cell(6).GetString().Should().BeEmpty();
+                row.Cell(7).GetString().Should().BeEmpty();
+                row.Cell(8).GetString().Should().BeEmpty();
+                row.Cell(9).GetString().Should().BeEmpty();
+                row.Cell(10).GetString().Should().BeEmpty();
+                row.Cell(11).GetString().Should().BeEmpty();
+                row.Cell(12).GetString().Should().Be("True");
+                row.Cell(13).GetString().Should().Be("Test@Test");
+            }
+        }
+
+        [Test]
+        public void OpenDelegatesTable_throws_exception_if_headers_dont_match()
+        {
+            // Given
+            var workbook = new XLWorkbook();

```

Comment:
Could we pull some of this into a "CreateWorkbookWithInvalidHeaders"? Or just add a workbook with invalid headers and use that